src/conversions/anyhow.rs

```diff
@@ -111,13 +111,21 @@ use crate::exceptions::PyRuntimeError;
 use crate::PyErr;
 
 impl From<anyhow::Error> for PyErr {
-    fn from(err: anyhow::Error) -> Self {
-        PyRuntimeError::new_err(format!("{:?}", err))
+    fn from(mut error: anyhow::Error) -> Self {
+        // Error chains & error with contexts are skipped
+        if error.source().is_none() {
+            error = match error.downcast::<Self>() {
```

Comment:
Rather than the assignment here, might it be simpler to just do `if let Ok(py_err) = error.downcast::<Self>() { return py_err; }`

And same for the eyre implementation.

Reply:
This does not work as `downcast` consumes `error`, but if the downcast fails, we still want to format it into a string. So either we duplicate that code path or use the assignment. Personally, I find the assignment to be an elegant solution to this problem.

Reply:
Ah great point, of course. Ignore me!